assets/helpers/payPalExpressCheckout/payPalExpressCheckout.js

```diff
@@ -113,15 +116,43 @@ function setup(dispatch: Function, getState: () => CombinedState, callback: Func
         commit: true,
 
         // This function is called when user clicks the PayPal button.
-        payment: setupPayment(dispatch, getState()),
+
+        payment: () => {
+
+          const htmlElement = document.getElementById(payPalId);
+
+          const dataset = htmlElement ? htmlElement.dataset : {};
+
+          const currentAmount = Number(dataset.amount) || amount;
+          const currentCurrency = ((dataset.currency: any): IsoCurrency) || currency;
+          const currentBillingPeriod = dataset.billingPeriod || billingPeriod;
+          const currentCsrfToken = dataset.csrfToken || csrfToken;
+
+          return setupPayment(
+            currentAmount,
+            currentBillingPeriod,
+            currentCurrency,
+            currentCsrfToken,
+          ).catch((err) => {
+            failureCallback(err);
+            throw err;
```

Comment:
Where would this error be caught and handled?

Reply:
This is the function that paypal call, and they expect a Promise so I think they handle the failure